CMakeLists.txt

```diff
@@ -274,8 +274,28 @@ if(CMAKE_C_COMPILER_ID MATCHES "GNU|AppleClang|Clang")
         "The compiler ${CMAKE_C_COMPILER_ID} does not support -fvisibility=hidden"
     )
   endif(NOT CC_SUPPORTS_VISIBILITY_HIDDEN)
+  # security options go here - these are not supported by all compilers
+  # first and foremost, do we have LTO?
+  check_c_compiler_flag(-flto CC_SUPPORTS_LTO)
```

Comment:
I think you can be pretty sure that if the `-fsanitize=cfi` works, `-flto`  will work as well, so no need to check this separately unless you want to provide a very specific error message saying what is wrong (and you do not have such a message here).

> To enable Clang’s available CFI schemes, use the flag -fsanitize=cfi. You can also enable a subset of available [schemes](https://clang.llvm.org/docs/ControlFlowIntegrity.html#cfi-schemes). As currently implemented, all schemes rely on link-time optimization (LTO); so it is required to specify -flto, and the linker used must support LTO, for example via the [gold plugin](https://llvm.org/docs/GoldPlugin.html).